pysb/core.py

```diff
@@ -1200,6 +1200,15 @@ def __init__(self, name, rule_expression, rate_forward, rate_reverse=None,
         self.move_connected = move_connected
         # TODO: ensure all numbered sites are referenced exactly twice within each of reactants and products
 
+        # Check synthesis products are concrete
+        if self.is_synth():
+            for cp in (self.reactant_pattern.complex_patterns if
```

Comment:
Suggest breaking this up a bit and using `any`:
```python
rp = self.reactant_pattern if self.is_reversible else self.product_pattern
if any(not cp.is_concrete() for cp in rp.complex_patterns):
    raise ...
```
That first line is pushing it on length, but even if that gets wrapped onto two lines I think it's more readable. Might want to wrap it so `rp` could be expanded to `reaction_pattern`.